src/sentry/api/endpoints/team_notification_settings_details.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any
+
+from rest_framework import status
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.bases.team import TeamEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.notification_setting import NotificationSettingsSerializer
+from sentry.api.validators.notifications import validate_type_option
+from sentry.models.notificationsetting import NotificationSetting
+
+
+class TeamNotificationSettingsDetailsEndpoint(TeamEndpoint):
+    """
+    This Notification Settings endpoint is the generic way to interact with the
+    NotificationSettings table via the API.
+    """
+
+    def get(self, request: Any, team: Any) -> Response:
+        """
+        Get the Notification Settings for a given User.
+        ````````````````````````````````
+        :pparam string team_slug: The slug of the team to get.
+        :qparam string type: If set, filter the NotificationSettings to this type.
+        :auth required:
+        """
+        if not features.has("organizations:incidents", team.organization, actor=request.user):
```

Comment:
why is this checking for the incidents feature flag?

Reply:
Oops that's a bug. I think I didn't have the feature flag name when I started this ticket.